File: app/src/main/java/com/example/gustavor/login/activities/RegisterActivity.java
package com.example.gustavor.login.activities;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.gustavor.login.utils.LoginUtils;
import com.example.gustavor.login.R;
import com.example.gustavor.login.daos.UserDao;

public class RegisterActivity extends AppCompatActivity {

    private EditText mUsername, mEmail, mPassword, mConfirmP;
    private Button mValidate, mCancel;
    private UserDao userDao;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);
        final Toolbar myToolbar = (Toolbar) findViewById(R.id.my_toolbar);
        setSupportActionBar(myToolbar);
        mUsername = (EditText) findViewById(R.id.username);
        mEmail = (EditText) findViewById(R.id.email);
        mPassword = (EditText) findViewById(R.id.password);
        mConfirmP = (EditText) findViewById(R.id.confirmP);
        mValidate = (Button) findViewById(R.id.confirmB);
        mCancel = (Button) findViewById(R.id.cancel);
        userDao = new UserDao(getApplicationContext());
        mValidate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (validateFields()) {
                    if(!userDao.insertUser(mUsername.getText().toString(), mEmail.getText().toString(), mPassword.getText().toString())){
                        Toast toast = Toast.makeText(getApplicationContext(), getString(R.string.repetitive_user), Toast.LENGTH_SHORT);
                        toast.show();
                    } else {
                        LoginUtils loginUtils = new LoginUtils(getApplicationContext());
                        loginUtils.saveLogin(mUsername.getText().toString(), userDao.getUser(mUsername.getText().toString()).getmId());
                        Toast toast = Toast.makeText(getApplicationContext(), getString(R.string.registered_login) + " " + mUsername.getText().toString() + "!", Toast.LENGTH_SHORT);
                        toast.show();
                        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                        startActivity(intent);
                    }
                }
            }
        });

        mCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
    }
    //pode alterar esse metodo para que ele se encaixe a suas necessidades de persistencia de dados
    private boolean validateFields() {
        if (mUsername.getText().toString().isEmpty()) {
            //setError e requestFocus são otimos para alertar o usuario de que algo esta errado
            mUsername.setError(getString(R.string.username_error));
            mUsername.requestFocus();
            return false;
        } else if (mEmail.getText().toString().isEmpty()) {
            mEmail.setError(getString(R.string.email_error));
            mEmail.requestFocus();
            return false;
        } else if (mPassword.getText().toString().isEmpty()) {
            mPassword.setError(getString(R.string.password_error));
            mPassword.requestFocus();
            return false;
        } else if (!mPassword.getText().toString().equals(mConfirmP.getText().toString())){
            mConfirmP.setError(getString(R.string.confirm_password_error));
            mConfirmP.requestFocus();
            return false;
        }
        return true;
    }
}

File: app/src/main/java/com/example/gustavor/login/RegisterActivity.kt
package com.example.gustavor.login

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_register.*

class RegisterActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)
        setSupportActionBar(my_toolbar as Toolbar)
        val userDao = UserDao(applicationContext)
        //referencia direta ao ID da view hehe
        confirmB.setOnClickListener {
            if (validateFields()) {
                //salva o usuario do cara no SharedPreferences e tambem o salva como "logado"
                val loginUtils = LoginUtils(applicationContext)
                loginUtils.saveLogin(username.text.toString())
                //inserindo no db
                userDao!!.insertUser(username.text.toString(), email.text.toString(), password.text.toString())
                val toast = Toast.makeText(applicationContext, getString(R.string.registered_login) + " " + username.text.toString() + "!", Toast.LENGTH_SHORT)
                toast.show()
                val intent = Intent(applicationContext, MainActivity::class.java)
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
                startActivity(intent)
            }
        }

        cancel.setOnClickListener { onBackPressed() }
    }

    //pode alterar esse metodo para que ele se encaixe a suas necessidades de persistencia de dados
    private fun validateFields(): Boolean {
        if (username.text.toString().isEmpty()) {
            //setError e requestFocus são otimos para alertar o usuario de que algo esta errado
            username.error = getString(R.string.username_error)
            username.requestFocus()
            return false
        } else if (email.text.toString().isEmpty()) {
            email.error = getString(R.string.email_error)
            email.requestFocus()
            return false
        } else if (password.text.toString().isEmpty()) {
            password.error = getString(R.string.password_error)
            password.requestFocus()
            return false
        } else if (password.text.toString() != confirmP.text.toString()) {
            confirmP.error = getString(R.string.confirm_password_error)
            confirmP.requestFocus()
            return false
        }
        return true
    }
}
